crates/ghost_actor/src/ghost_channel.rs

```diff
@@ -452,11 +456,26 @@ impl<
                         request_id,
                         payload,
                     } => {
+                        let responder: GhostResponder<RequestToSelfResponse, Error> = match request_id {
```

Comment:
So it looks like this makes the `respond()` method on GhostMessage obsolete? Maybe we should move this code into the GhostMessage and convert `respond()` into `make_responder()`?